wasm/src/lib.rs

```diff
@@ -58,7 +58,20 @@ fn oom(layout: ::core::alloc::Layout) -> ! {
 
 #[no_mangle]
 extern "C" fn _iroha_wasm_alloc(len: usize) -> *const u8 {
-    core::mem::ManuallyDrop::new(Vec::<u8>::with_capacity(len)).as_mut_ptr()
+    if len == 0 {
+        debug::dbg_panic("Cannot allocate 0 bytes");
+    }
+    let layout = alloc::alloc::Layout::array::<u8>(len).dbg_expect("Cannot allocate layout");
+    // Safety: safe until `layout` has non-zero size
```

Comment:
I think in `Safety` section we should specify, why using unsafe is safe in this case e.g.

```rust
// Safety: safe because `layout` has non-zero size
```

So if we write `unsafe` trait or function we write what invariants should hold and in `unsafe` block we write why this invariants hold in this particular case.

Reply:
Right